inc/vcf/assembly_report_writer.hpp

```diff
@@ -63,21 +71,68 @@ namespace ebi
       public:
         SummaryAssemblyReportWriter(){}
 
-        void write_mismatch(const vcf::RecordCore &record_core)
+        ~SummaryAssemblyReportWriter()
+        {
+
+        }
+
+        void match(const vcf::RecordCore &record_core, std::string & match_result) override
+        {
+            match_stats.add_match_result(true);
+        }
+
+        void mismatch(const vcf::RecordCore &record_core, std::string &mismatch_result) override
         {
             match_stats.add_match_result(false);
         }
 
-        void write_match(const vcf::RecordCore &record_core)
+        virtual void write_warning(const vcf::RecordCore &record_core, std::string &warning) override
+        {
+
+        }
+    };
+
+    class ValidAssemblyReportWriter : public AssemblyReportWriter
+    {
+      public:
+        ValidAssemblyReportWriter(std::string filename) : file_name(filename)
+        {
+            file.open(filename, std::ios::out);
+            if(!file) {
+                throw std::runtime_error{"Unable to write output file " + file_name};
+            }
+        }
+
+        ~ValidAssemblyReportWriter()
+        {
+            file.close();
+        }
+
+        virtual void match(const vcf::RecordCore &record_core, std::string &match_result) override
         {
             match_stats.add_match_result(true);
+            file << record_core.line;
```

Comment:
Even though we aimed to merge both `RecordCore` classes, the idea is to drop this line as an attribute, otherwise the `RecordCore` class won't save any memory compared to the `Record` class. The line can be made an argument for the assembly report writers instead.